src/linear_algebra/finite_dimensional.lean

```diff
@@ -1197,8 +1208,96 @@ lemma set_is_basis_of_linear_independent_of_card_eq_finrank
   is_basis K (coe : s → V) :=
 is_basis_of_linear_independent_of_card_eq_finrank lin_ind (trans s.to_finset_card.symm card_eq)
 
+lemma singleton_is_basis (v : V) (nz : v ≠ 0) (h : finrank K V = 1) :
+  is_basis K (coe : ({v} : set V) → V) :=
+set_is_basis_of_linear_independent_of_card_eq_finrank
+  (linear_independent_singleton nz) (by simp [h])
+
 end is_basis
 
+/-!
+We now give characterisations of `finrank K V = 1` and `finrank K V ≤ 1`.
+-/
+section finrank_eq_one
+
+/-- If there is a nonzero vector and every other vector is a multiple of it,
+then the module has dimensions one. -/
+lemma finrank_eq_one (v : V) (n : v ≠ 0) (h : ∀ w : V, ∃ c : K, c • v = w) :
+  finrank K V = 1 :=
+begin
+  convert finrank_eq_card_basis ((is_basis_singleton_iff punit v).mpr _),
+  exact ⟨n, h⟩,
+end
+
+/--
+If every vector is a multiple of some `v : V`, then `V` has dimension at most one.
+-/
+lemma finrank_le_one (v : V) (h : ∀ w : V, ∃ c : K, c • v = w) :
+  finrank K V ≤ 1 :=
+begin
+  by_cases n : v = 0,
+  { subst n,
+    convert zero_le_one,
+    haveI := subsingleton_of_forall_eq (0 : V) (λ w, by { obtain ⟨c, rfl⟩ := h w, simp, }),
+    exact finrank_zero_of_subsingleton, },
+  { exact (finrank_eq_one v n h).le, }
+end
+
+/--
+A module with a nonzero vector `v` has dimension 1 iff `{v}` is a basis.
+-/
+lemma finrank_eq_one_iff_of_nonzero (v : V) (nz : v ≠ 0) :
+  finrank K V = 1 ↔ is_basis K (λ x : ({v} : set V), (x : V)) :=
+⟨singleton_is_basis v nz, λ b, by convert finrank_eq_card_basis b⟩
```

Comment:
Since the `linear_independent (λ x : ({v} : set V), (x : V))` half of `is_basis` always holds in this case (by `linear_independent_singleton nz`), wouldn't it be a better theorem that `finrank K V = 1` iff `v` spans the whole space `V`?

Reply:
Good suggestion. I've implemented this now.